fiftyone/core/sample.py

```diff
@@ -464,6 +464,9 @@ def save(self):
                 "Cannot save a sample that has not been added to a dataset"
             )
 
+        if self._dataset.allow_sample_save_batching:
+            return self._dataset.batch_save_sample(self)
```

Comment:
If you track down how the document is saved in non-batching mode, you get to this code:

https://github.com/voxel51/fiftyone/blob/1e276f32d2b9b9242f3a6fde3f6a8bfa37aefa4c/fiftyone/core/odm/document.py#L359-L478

Does `doc._get_update_doc()` do everything that's done there in terms of clearing changed fields, etc?

Side note: I believe this code was copied from MongoEngine into our repo by a previous engineer. I don't know why, but presumably they wanted to tweak something about the behavior? Not sure...

Reply:
It was just a method I found that seemed to fit the need. If there's a preferred way to get the update doc I'm open to it.

Reply:
I don't have a "preference" per se, just pointing out that different code is being run to generate the updates now depending on code path

Reply:
I see now that to save for frames and views I'm going to have to change my approach. I might do a little abstracting on this class to be able to pull out the guts of a save call that way I can aggregate them.